common/remote_object.go

```diff
@@ -40,17 +39,62 @@ func (pe *objectPropertyParseError) Unwrap() error {
 	return pe.error
 }
 
+type multiError struct {
```

Comment:
Would it make more sense to move this to `multierror.go` file? Instead of overcrowding this file or the already crowded `helpers.go` file. Also, IMO, having a test for this in `remote_object_test.go` file does not make sense.

Reply:
I was thinking the same, but then I realised that `multierror` is only used in one file, so it makes sense to leave the type close to where it's used.

Reply:
Yes, I thought that too, but as a helper, it still makes more sense to me in a new file. Also, in Go 1.20+, we will remove this file as there are multiple errors support is on the way. Having it in a new file will also make us spot it more easily.

Reply:
Exactly the multierror implementation is only used in this file, and I would keep its definition inside this file only and try to not widespread its usage. Actually after the merge is done I would create an issue in order to get rid of this implementation or replace it by something simpler.

Reply:
My reason to moving it to a new file would be if we think that there is a valid use case for "multierrors" and therefore want to use them in more places. In that case I think it's good to move it as a helper and improve its implementation in future releases. Otherwise I would keep it in this file and try to remove it for something simpler.

What do you think?

Reply:
@ka3de I would rather create that issue soon without waiting for the merge.

Reply:
Done! See #738 .

Reply:
Thanks!